rl_zoo3/tracking/git.py

```diff
@@ -0,0 +1,62 @@
+import os
+from git import Repo, InvalidGitRepositoryError, GitError
+from pathlib import Path
+from typing import Dict
+import logging
+
+IGNORE_REPO_FILENAME = "DONT_TRACK"
+
+logger = logging.getLogger(__name__)
+
+
+class UncommittedChangesError(GitError):
+    pass
+
+
+def track_git_repos(working_path: Path | None = None) -> Dict[str, str]:
+    """
+    Tracks Git repositories within a working directory and logs their latest commit hashes.
+
+    Raises `UncommittedChangesError` if any repositories have uncommitted changes.
+
+    Repositories can be excluded from version checking by creating an empty file named
+    "DONT_TRACK" in their subfolders.
+
+    Args:
+        working_path (pathlib.Path, optional): The starting path. If None, the
+            current directory is assumed.
+
+    Returns:
+        A dictionary mapping repository folder names (str) to their latest commit hashes (str).
+    """
+
+    git_repos_commits = {}
+    if working_path is None:
+        working_path = _get_current_working_dir()
```

Comment:
Do you want to define _get_current_working_dir before calling it?

Reply:
Good question. AFAIK there is no official convention within Python itself ( https://stackoverflow.com/questions/30847925/order-of-functions-within-a-python-source-file ).

I was trying to achieve a document, which could be read from the top to the bottom, following the guidelines in _Clean Code_ by Robert C. Martin.

Defining `_get_current_working_dir()` after `track_git_repos()` is intentional - the implementation behind that should not be important as long as the function name is self explanatory.